File: Senior Yr/Fall/cs380/Assignments/Assignment3/src/SlidingBrickPuzzle/Node.java
/*
 * To make life easier I made varibles public
 */
package SlidingBrickPuzzle;

import java.util.ArrayList;

/**
 *
 * @author sunnypatel
 */
public class Node {
    public Node parent;
    public ArrayList<Node> children;
    public Move move;
    public int depth;
    public double gFunc;
    public Puzzle state;
    
    public Node(Node parent){
        this.parent = parent;
        this.depth = parent.depth + 1;
        this.children = new ArrayList<Node>();
    }
    
    public Node(Puzzle state){
        this.state = state;
        this.parent = null;
        this.depth = 0;
        this.children = new ArrayList<Node>();
    }
    
    public void addChild(Node child){
        this.children.add(child);
    }
    
    public void setState(Puzzle state){
        this.state = state;
    }

    public void setMove(Move m){
        this.move = m;
    }
}

File: Senior Yr/Fall/cs457/PA2-tests/svp28-PA2/Node.cpp
#include "Node.h"

Node::Node(){
	parent = NULL;
	h=0;
	g=0;
	depth = 0;
}

Node::Node(Node* parentTmp){
	parent = parentTmp;
	h = 0;
	g = 0;
	depth = 0;
}

Node::Node(const Node& orig){
	parent = orig.parent;
    children = orig.children;
    move = orig.move;
    depth = orig.depth;
    g = orig.g;
    h = orig.h;
    hash = orig.hash;
    state = orig.state;
   movesSoFar = orig.movesSoFar;
}

Node::~Node(){

}

void Node::setParent(Node p){
	parent = &p;
}

void Node::addMove(Move move){
	movesSoFar.push_back(move);
}

void Node::addChild(Node child){
	
	children.push_back(child);
}

bool Node::equals(Node* n){
	if((n->move.equals(&this->move)) && (this->state.compare(&n->state)))
		return true;

	return false;
}